multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/jobs/ConfigurationSubscriptionCleaner.java

```diff
@@ -0,0 +1,105 @@
+import org.cloudfoundry.multiapps.controller.core.util.SecurityUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.HttpStatus;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+@Named
+@Order(40)
+public class ConfigurationSubscriptionCleaner implements Cleaner {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfigurationSubscriptionCleaner.class);
+    private static final String AUTH_ORIGIN = "uaa";
+
+    protected ApplicationConfiguration configuration;
+    protected ConfigurationSubscriptionService configurationSubscriptionService;
+    protected CloudControllerClient cfClient;
+    private boolean executed;
+
+    @Inject
+    public ConfigurationSubscriptionCleaner(ApplicationConfiguration applicationConfiguration, ConfigurationSubscriptionService configurationSubscriptionService) {
+        this.configuration = applicationConfiguration;
+        this.configurationSubscriptionService = configurationSubscriptionService;
+        this.executed = false;
+    }
+
+    @Override
+    public void execute(Date expirationTime) {
+        if (!executed) {
+            LOGGER.debug(CleanUpJob.LOG_MARKER, "Deleting orphaned configuration subscriptions...");
+            deleteOrphanedConfigurationSubscriptions();
+            LOGGER.debug(CleanUpJob.LOG_MARKER, "Orphaned configuration subscriptions deleted");
+            executed = true;
+        }
+    }
+
+    private void deleteOrphanedConfigurationSubscriptions() {
+        List<ConfigurationSubscription> configurationSubscriptions = configurationSubscriptionService.createQuery()
+                                                                                                     .list();
+        configurationSubscriptions.stream()
+                                  .filter(this::hasNoAssociatedSpace)
+                                  .peek(this::auditLogDeletion)
+                                  .map(ConfigurationSubscription::getSpaceId)
+                                  .distinct()
+                                  .forEach(this::deleteConfigurationSubscriptionsBySpaceId);
+    }
+
+    private boolean hasNoAssociatedSpace(ConfigurationSubscription configurationSubscription) {
+        String spaceId = configurationSubscription.getSpaceId();
+        return !spaceExists(spaceId);
+    }
+
+    private boolean spaceExists(String spaceId) {
+        if (cfClient == null) {
+            initCloudControllerClient();
+        }
+        try {
+            cfClient.getSpace(UUID.fromString(spaceId));
+            return true;
+        } catch (CloudOperationException e) {
+            if (e.getStatusCode()
+                 .equals(HttpStatus.NOT_FOUND)) {
+                return false;
+            }
+            throw e;
```

Comment:
If one exception is thrown at the begining of stream processing, the deletion of other entries will not happen at all.
So, let's not rethrow the exception.

Reply:
only logging as a handling?